src/main/java/org/traccar/protocol/ZrProtocolDecoder.java

```diff
@@ -0,0 +1,323 @@
+        }
+    }
+
+    private static void sendLbsTransferResponse(Channel channel, SocketAddress remoteAddress, ByteBuf id, int frameType, short editionNum, byte encryptionType, Integer packageNo) {
+        ByteBuf response = Unpooled.buffer();
+        add2391(response);
+        add2094(response);
+        add23A3(response);
+        if (channel != null) {
+            channel.writeAndFlush(new NetworkMessage(formatMessage(FRAME_TYPE_LBS_TRANSFER_LONG_LAT_ACK, id, editionNum, encryptionType, packageNo, response), remoteAddress));
+        }
+    }
+
+    private static void sendTimeSynResponse(Channel channel, SocketAddress remoteAddress, DeviceSession deviceSession, ByteBuf id, int frameType, short editionNum, byte encryptionType, Integer packageNo) {
+        ByteBuf response = Unpooled.buffer();
+        add2391(response);
+        add2510(response, deviceSession);
+        if (channel != null) {
+            channel.writeAndFlush(new NetworkMessage(formatMessage(FRAME_TYPE_TIME_SYN_ACK, id, editionNum, encryptionType, packageNo, response), remoteAddress));
+        }
+    }
+
+    public static void add2391(ByteBuf bodyBuf) {
+        bodyBuf.writeShort(0x2391);
+        bodyBuf.writeShort(3);
+        bodyBuf.writeBytes(ByteBufUtil.decodeHexDump("123456"));
+    }
+
+    private static void add23A0(ByteBuf bodyBuf, int frameType, Integer packageNo) {
+        bodyBuf.writeShort(0x23a0);
+        bodyBuf.writeShort(7); // len
+
+        bodyBuf.writeShort(packageNo);
+        bodyBuf.writeByte(0);
+        bodyBuf.writeShort(frameType);
+        bodyBuf.writeShort(0);
+    }
+
+    private static void add23A3(ByteBuf bodyBuf) {
+        bodyBuf.writeShort(0x23a3);
+        bodyBuf.writeShort(4);
+        long time = System.currentTimeMillis() / 1000;
+        bodyBuf.writeInt((int) time);
+    }
+
+    private static void add2094(ByteBuf bodyBuf) {
+        bodyBuf.writeShort(0x2094);
+        bodyBuf.writeShort(8);
+        bodyBuf.writeInt(0); // longitude
+        bodyBuf.writeInt(0); // latitude
+    }
+
+    private static void add2510(ByteBuf bodyBuf, DeviceSession deviceSession) {
+        bodyBuf.writeShort(0x2510);
+        bodyBuf.writeShort(9);
+        TimeZone timeZone = deviceSession.get(DeviceSession.KEY_TIMEZONE);
+        int timeZoneOffset = timeZone.getRawOffset() / 3600000;
+        bodyBuf.writeByte(timeZoneOffset);
+        bodyBuf.writeByte(1);
+
+        Calendar calendar = Calendar.getInstance(timeZone);
+        bodyBuf.writeShort(calendar.get(Calendar.YEAR));
+        bodyBuf.writeByte(calendar.get(Calendar.MONTH) + 1);
+        bodyBuf.writeByte(calendar.get(Calendar.DAY_OF_MONTH));
+        bodyBuf.writeByte(calendar.get(Calendar.HOUR_OF_DAY));
+        bodyBuf.writeByte(calendar.get(Calendar.MINUTE));
+        bodyBuf.writeByte(calendar.get(Calendar.SECOND));
+    }
+
+    /**
+     * bcc check
```

Comment:
What is bcc? Is it a standard algorithm? I feel like you should be able to use one of our existing implementations.